client/src/main/scala/com/dwolla/cloudflare/domain/model/accesscontrolrules/package.scala

```diff
@@ -0,0 +1,50 @@
+package com.dwolla.cloudflare.domain.model
```

Comment:
Removing `AccessControlRule`'s usage of DTOs and consolidating domain related logic into a package object aligns it with other recently added `scala-cloudflare` resources (such as rate limits and page rules). It also makes integration with `cloudflare-lambda` simpler as the same domain object (`AccessControlRule`) can be used when handling all cloudflare request types.